sdk/framework/openapi.go

```diff
@@ -340,8 +328,12 @@ func documentPath(p *Path, backend *Backend, requestResponsePrefix string, doc *
 			op.Deprecated = props.Deprecated
 			op.OperationID = operationID
 
-			// Add any fields not present in the path as body parameters for POST.
-			if opType == logical.CreateOperation || opType == logical.UpdateOperation {
+			switch opType {
+			// For the operation types which map to POST/PUT methods, and so allow for request body parameters,
+			// prepare the request body definition
+			case logical.CreateOperation:
+				fallthrough
+			case logical.UpdateOperation:
```

Comment:
Note for reviewer: Swapping `if` for `switch` because I need to increase the number of cases, and can make good use of `fallthrough`.